openwisp_utils/bots/ci_failure/bot.py

```
@@ -0,0 +1,452 @@
+        try:
+            pr_num = int(self.pr_number)
+        except ValueError as e:
+            print(f"Invalid PR number: {e}")
+            return None
+        try:
+            pr = self.repo.get_pull(pr_num)
+        except GithubException as e:
+            print(f"Error fetching PR: {e}")
+            return None
+        try:
+            result = subprocess.run(
```

Ruff (0.14.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)